zigpy/types/basic.py

```diff
@@ -150,9 +153,17 @@ def serialize(self):
 
     @classmethod
     def deserialize(cls, data):
-        bytes = int.from_bytes(data[:cls._prefix_length], 'little')
-        s = data[cls._prefix_length:bytes + 1]
-        return cls(s), data[bytes + 1:]
+        if len(data) < cls._prefix_length:
```

Comment:
`if len(data) < cls._prefix_length:`
Nah, this is wrong. `cls._prefix_length` specifies the size in bytes (uint8_t or uint16_t) for the field containing the "length" of data. eg the prefix length of 1 == 1 byte means the length of data should be 0<= len(data) <= 254 (255 is considered invalid length)

Also I could have zero data eg `b''` so `(len(data < cls._prefix_length) is True` while correctly it should deserialize into `b\x00`

Reply:
ugh, it is too late. I confused deserialize vs serialize

Reply:
I see what you did here. Nice. 
Do you actually see devices sending the wrong data? ie not enough data to deserialize?

Reply:
No, not yet. I was converting some OTA code to use the zigpy types module and noticed it might see the same hard-to-debug issue I ran into with similar code using slicing and `int.from_bytes` for parsing.